### ISSUE TITLE
Renaming window titles

### ISSUE THREAD

**Issue by Fan-iX** [netxs-group/vtm#430]
I can change the title of a terminal window by setting the xterm title using `echo -ne "\e]0;some title\007"`, so that I can distinguish between different terminal windows.
Is it possible to rename (preferably edit) other windows like Tile Manager? Now all tiling windows are shown as "Tiling Window Manager" so that I cannot distinguish them clearly before I activate them.

**Comment by o-sdn-o** [comment]
Indeed, we need to add the ability to edit window titles in the list of windows in the taskbar or in the window itself.

**Comment by o-sdn-o** [comment]
I implemented renaming the tile manager title using the window menu button. By clicking on this button, the window title changes using the clipboard:

https://github.com/netxs-group/vtm/assets/11535558/e12aa895-15e4-46c9-a4a7-d1f72f2090c0

**Comment by mlopezgva** [comment]
Hi!

Doesn't work for me. Using vtm on wezterm on Linux Mint Cinnamon. that would be most interesting.

**Comment by o-sdn-o** [comment]
Hi! On Unix-like systems there is no direct access to the system clipboard yet, and you may need to copy text inside vtm first to get the text into vtm's internal clipboard.

**Comment by o-sdn-o** [comment]
If this approach doesn't work, then maybe I broke something and I'll fix it.

**Comment by o-sdn-o** [comment]
In the upcoming release, it will be possible to do this using a keyboard shortcut. The keyboard shortcut can be specified in the settings.

**Comment by mlopezgva** [comment]
> Hi! On Unix-like systems there is no direct access to the system clipboard yet, and you may need to copy text inside vtm first to get the text into vtm's internal clipboard.

This one works. Just type what I want on the shell and copy it directly. Thanks.

Any way to rename "Site" _'apps'_? Right now they're not very useful if I do not know what they "contain". :wink: 

I gather it's not possible to launch, say, btop, htop or mc directly in their own "window", since they do not create the tty..?

Launching them (like I'd do with `tmux -n TOP btop`), even in a tile app, would be most interesting. But that's out of the scope of this issue... :angel:

**Comment by o-sdn-o** [comment]
> Any way to rename "Site" 'apps'? Right now they're not very useful if I do not know what they "contain". 😉

Right clicking on 'Site' will set its title using the internal clipboard.

**Comment by o-sdn-o** [comment]
Oh, I see now that the script engine has been broken for some time now and it is impossible to launch anything through it. I will fix it in upcoming release v0.9.99.56.

> I gather it's not possible to launch, say, btop, htop or mc directly in their own "window", since they do not create the tty..?

Run a CUI application in the running desktop:
```
echo "vtm.run(title='MC' cmd='mc')" | vtm
```

> Launching them (like I'd do with tmux -n TOP btop), even in a tile app, would be most interesting. But that's out of the scope of this issue... 👼

Run Tiling Window Manager with three terminals attached:
```
echo "vtm.run(type=tile title=Terminals cmd='v(h(Term,Term),Term)')" | vtm
```

**Comment by o-sdn-o** [comment]
> Launching them (like I'd do with tmux -n TOP btop), even in a tile app, would be most interesting.


Run the following inside vtm desktop (since v0.9.99.56):
```
echo "vtm.set(id=mc title='MC' cmd='mc')" | vtm
echo "vtm.set(id=htop title='HTop' cmd='htop')" | vtm
echo "vtm.run(type=tile title=Windows cmd='v(h(mc,htop),mc)')" | vtm
```

**Comment by Fan-iX** [comment]
Hi, I can't get the script engine to work correctly in my Ubuntu-22.04 WSL system. `vtm.run(type=term cmd=...)`, `vtm.run(type=dtvt cmd=...)` and `vtm.run(type=term, cmd=...)` all create a blank window on the vtm desktop, no matter what `cmd` I give. 
![image](https://github.com/user-attachments/assets/8d11555b-d4c5-455e-9bd1-0671533cae0e)
However, `vtm.run(type=tile)` work as expected.

**Comment by o-sdn-o** [comment]
Apparently `type=term` is broken somewhere. I'll fix it. As a workaround you can try using `type=terminal` instead.

**Comment by o-sdn-o** [comment]
This issue affected all cases where the `cwd=` parameter was used (including implicitly, as in the case of vtm.run(...)). The message about a successful change of the current directory was written in plain text to the dtvt binary stream, thus breaking communication.
